platform/mbed-trace/CMakeLists.txt

```diff
@@ -1,17 +1,23 @@
-INCLUDE(CMakeForceCompiler)
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
 
-cmake_minimum_required (VERSION 2.8)
-SET(CMAKE_SYSTEM_NAME Generic)
+if(DEFINED TARGET_LIKE_X86_LINUX_NATIVE)
```

Comment:
Why make it specific to the x86 architecture? How about running on an Arm laptop/desktop/server?

Reply:
This one comes from external lib, it was consolidated, see https://github.com/ARMmbed/mbed-trace/blob/master/source/CMakeLists.txt. It looks like we changed it but we only consolidated the cmakelists as it was split into multiple. It might be better to just revert this and let it as it was, and be fixed later.

@hugueskamba is that correct? Can we revert this one?

Reply:
As mentioned by @0xc0170 ,
This is a consolidation of what is upstream.
Please see this PR that reverts the consolidation: https://github.com/ARMmbed/mbed-os/pull/13590